azure/durable_functions/decorators/decorators.py

```diff
@@ -0,0 +1,128 @@
+from triggers import OrchestrationTrigger, ActivityTrigger, EntityTrigger, EntityClient, OrchestrationClient, DurableClient
+from typing import Callable, Dict, List, Optional
+from entity import Entity
+from orchestrator import Orchestrator
+
+class FunctionApp: # TODO: this doesn't seem right
```

Comment:
Where do we want these decorators to live? Don't we have to inherit from some particular `azure.functions` class? If so - which is it? In other words - how do I register these decorators with the PyStein system?

Reply:
Hi David, as blueprint change is in, you can create a new toplevel class ```class DurableFunctionApp(FunctionRegister, TriggerApi, BindingApi)```  (not sure if BindingApi is needed, plz see if any methods in BindingApi class in lib is used in durable) containing all the decorator methods. User will create it to use the decorators on their methods.

```
class DurableFunctionApp(FunctionRegister, TriggerApi, BindingApi):
        def on_orchestration_change(self, name: str, orchestration: Optional[str] = None):
        .......

durableFuncApp = DurableFunctionApp(params)

@durableFuncApp.on_orchestration_change(xxxx)
def user_func():
    ....
```

Reply:
Thanks, I made this subclass from FunctionRegister and TriggerApi.
From concept alone, I'm fairly certain we don't utilize the BindingApi, but we also don't manually call any methods from TriggerApi so I'm not sure.

What's an example of a TriggerApi that we would/could invoke?